src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -282,7 +282,7 @@ public override Task FlushAsync()
                 throw new ObjectDisposedException("stream");
             }
 
-            return FlushInternalAsync();
+            return FlushInternalAsync(flushEncoder: true);
```

Comment:
>This writer can certainly be used after either of those calls.

That's not really related though.

flushEncoder has to do with the behavior of surrogate pairs.

The encoder buffers the first `char` if it's a surrogate pair in expectation of seeing the second part of the pair. Since we want to write everything buffered when you call `Flush()` it should be true in that case. 

In the other cases, we might have more output, so it's technically more correct to leave it false.

This likely has no impact on anything since we always call flush. I accidentally changed this behavior about a week ago so I'm putting it back to what it was.